### DISCUSSION TITLE
怎么显示文件的sha1和sha256？

### DISCUSSION THREAD

**Discussion by wc7086** [spencerwooo/onedrive-vercel-index#457] [Q&A] [answered]
现在还没有这个功能吗？未来打算如何实现？

**Comment by spencerwooo** [comment, marked as answer]
无法预览的文件（比如 `.tar` 压缩包），点击会默认显示文件的各项属性。示例链接：

<pre>
<a href="https://drive.swo.moe/%F0%9F%A5%9F%20Some%20test%20files/Compressed/spaces.tar" title="https://drive.swo.moe/%F0%9F%A5%9F%20Some%20test%20files/Compressed/spaces.tar">https://drive.swo.moe/%F0%9F%A5%9F%20Some%20test%20files/Compressed/spaces.tar</a>
</pre>

OneDrive 国际版支持所有参数，但是 E5 开发者版（以及 OneDrive for Business/Edu 我猜测）不支持显示，因为上游 OneDrive API 不返回相关参数。

**Reply by wc7086** [reply]
能做到读取同目录下同名的的 .sha256 和 .sha1 后缀的文件来显示 sha256 和 sha1 值吗？

**Reply by myl7** [reply]
有什么 RFC 或是软件实现支持这种摆放方式吗？如果只是一种约定俗成的话感觉意义不大，毕竟可以直接写在 readme 里来展示。

**Reply by wc7086** [reply]
> 有什么 RFC 或是软件实现支持这种摆放方式吗？

https://superuser.com/questions/1566136/is-the-sha256-file-format-formally-defined-somewhere-how-should-it-be-parsed
https://superuser.com/questions/1312740/how-to-take-sha256sum-of-file-and-compare-to-check-in-one-line
https://www.gnu.org/software/coreutils/sha256sum

**Reply by myl7** [reply]
如果要做的话，可以：

- 文件预览页面内嵌显示
- 打开 hash 值文件时用单独的预览界面
- 打开 hash 值文件时当作 txt 预览

@spencerwooo 你觉得有必要吗？如果没有必要，那应该至少也可以把 md5 sha1 sha256 sha512 后缀的文件当做 txt 来预览？

**Reply by spencerwooo** [reply]
暂时不考虑。
